src/components/motor_remap/MotorRemapComponent.js

```diff
@@ -0,0 +1,220 @@
+'use strict';
+
+class MotorRemapComponent
+{
+    constructor(contentDiv, onLoadedCallback, droneConfiguration, motorStopValue, motorSpinValue)
+    {
+        this._contentDiv = contentDiv;
+        this._onLoadedCallback = onLoadedCallback;
+        this._droneConfiguration = droneConfiguration;
+        this._motorStopValue = motorStopValue;
+        this._motorSpinValue = motorSpinValue;
+        this._config = new MotorRemapConfig(100);
+
+        this._currentJerkingTimeout = -1;
+        this._currentJerkingMotor = -1;
+
+        this._currentSpinningMotor = -1;
+
+        this._contentDiv.load("./components/motor_remap/body.html", ()=>{ this._setupdialog(); });
+    }
+
+    _setupdialog()
+    {
+        i18n.localizePage();
+
+        this._resetGui();
+
+        $('#motorsEnableTestMode-dialogMotorRemap').change(function ()
+        {
+            var enabled = $(this).is(':checked');
+
+            if (enabled) {
+                $('#dialogMotorRemapAgreeButton').show();
+            } else {
+                $('#dialogMotorRemapAgreeButton').hide();
+            }
+        });
+
+        $('#dialogMotorRemapAgreeButton').click(()=>{ this._onAgreeButtonClicked(); });
+        $("#motorsRemapDialogStartOver").click(()=>{ this._startOver(); });
+        $("#motorsRemapDialogSave").click(()=>{ this._save(); });
```

Comment:
You will see that Sonar asks for replace some of the "strings" by constants if you use them more than twice. We sometimes ignore it (the code can be worse with some constants here), and when we want to "fix" it we simply define some global variables referencing this elements at the beginning (the constructor will be a good place to assign them), and we use them in the rest of the code.